### PR TITLE
String solver cbmc pass option

### PR DESCRIPTION
**Pull request by romainbrenguier**
Adding a PASS option for calling the string solver and changing the Makefile for this string solver

### CODE REVIEW

File: src/cbmc/cbmc_parse_options.cpp [hunk 1]
```diff
@@ -1175,6 +1187,7 @@ void cbmc_parse_optionst::help()
     " --yices                      use Yices\n"
     " --z3                         use Z3\n"
     " --refine                     use refinement procedure (experimental)\n"
+    " --pass                       use pass procedure (experimental)\n"
```
Comment (tautschnig) [inline src/cbmc/cbmc_parse_options.cpp]: So ... what is the pass procedure? Would be nice to be user-friendly in the --help output.

File: src/cbmc/cbmc_solvers.h [hunk 2]
```diff
@@ -79,11 +79,14 @@ class cbmc_solverst:public messaget
   virtual std::unique_ptr<solvert> get_solver()
   {
     solvert *solver;
-
     if(options.get_bool_option("dimacs"))
       solver = get_dimacs();
     else if(options.get_bool_option("refine")) 
       solver = get_bv_refinement();
+    else if(options.get_bool_option("pass")) {
+      std::cout << "PASS solver" << std::endl;
```
Comment (tautschnig) [inline src/cbmc/cbmc_solvers.h]: Use a proper message_streamt API

File: src/cbmc/cbmc_solvers.h [hunk 3]
```diff
@@ -79,11 +79,14 @@ class cbmc_solverst:public messaget
   virtual std::unique_ptr<solvert> get_solver()
   {
     solvert *solver;
-
```
Comment (tautschnig) [inline src/cbmc/cbmc_solvers.h]: Why remove whitespace?

File: src/cbmc/cbmc_solvers.cpp [hunk 4]
```diff
@@ -324,6 +325,29 @@ cbmc_solverst::solvert* cbmc_solverst::get_bv_refinement()
...
+ 
+cbmc_solverst::solvert* cbmc_solverst::get_string_refinement()
+{
+  propt *prop;
+  prop=new satcheck_no_simplifiert();
+  prop->set_message_handler(get_message_handler());
+
+  string_refinementt *string_refinement = new string_refinementt(ns, *prop);
+  string_refinement->set_ui(ui);
+  return new cbmc_solver_with_propt(string_refinement, prop);
```
Comment (tautschnig) [inline src/cbmc/cbmc_solvers.cpp]: That's two objects allocated on the heap and never freed?

File: src/cbmc/cbmc_parse_options.cpp [hunk 5]
```diff
@@ -341,6 +342,11 @@ void cbmc_parse_optionst::get_command_line_options(optionst &options)
     options.set_option("refine-arithmetic", true);
   }
 
+  if(cmdline.isset("pass"))
+  {
+    options.set_option("pass", true);
```
Comment (tautschnig) [inline src/cbmc/cbmc_parse_options.cpp]: How about `options.set_option("pass", cmdline.isset("pass"));` ?

File: src/cbmc/cbmc_parse_options.cpp [hunk 6]
```diff
@@ -911,6 +917,12 @@ bool cbmc_parse_optionst::process_goto_program(
     status() << "Partial Inlining" << eom;
     goto_partial_inline(goto_functions, ns, ui_message_handler);
     
+
+    if(cmdline.isset("pass")) {
```
Comment (tautschnig) [inline src/cbmc/cbmc_parse_options.cpp]: By the looks of it: is `options.set_option("pass", ...)` even required?

Comment (romainbrenguier) [inline src/cbmc/cbmc_parse_options.cpp]: It is used in `cbmc_parse_options.h`

**Comment by romainbrenguier** [comment]
this is made obsolete by the more recent pull request #341

**Pull request closed without merging**